test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -166,6 +166,80 @@ TEST_F(XdsOverrideHostTest, SubchannelsComeAndGo) {
   ExpectRoundRobinPicks(picker.get(), {kAddresses[1]}, call_attributes);
 }
 
+TEST_F(XdsOverrideHostTest, FailedSubchannelIsNotPicked) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto picker = ExpectStartupWithRoundRobin(kAddresses);
+  ASSERT_NE(picker, nullptr);
+  // Check that the host is overridden
+  std::map<UniqueTypeName, std::string> pick_arg{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), pick_arg), kAddresses[1]);
+  auto subchannel = FindSubchannel(kAddresses[1]);
+  ASSERT_NE(subchannel, nullptr);
+  subchannel->SetConnectivityState(GRPC_CHANNEL_TRANSIENT_FAILURE,
```

Comment:
A subchannel will not go from READY directly to TRANSIENT_FAILURE; instead, it will go into IDLE, then RR will ask it to reconnect, it will go into CONNECTING, and then if that connection attempt fails, it will go into TRANSIENT_FAILURE.